File: src/test/java/data/CalendarTableColumn.java
package data;

import lombok.Getter;

public enum CalendarTableColumn implements DataTableColumn<CalendarTableColumn> {
    TIME(    "Time,"),
    CURRENCY("Currency"),
    EVENT(   "Event"),
    ACTUAL(  "Actual"),
    FORECAST("Forecast"),
    PREVIOUS("Previous");

    @Getter
    final String title;

    CalendarTableColumn(String title) {
        this.title = title;
    }
}

File: src/test/kotlin/org/skefir/entity/CalendarTableColumn.kt
package org.skefir.entity



enum class CalendarTableColumn(private val title: String) : DataTableColumn {
    TIME(    "Time,"),
    CURRENCY("Currency"),
    EVENT(   "Event"),
    ACTUAL(  "Actual"),
    FORECAST("Forecast"),
    PREVIOUS("Previous");

    override fun getTitle(): String {
        return title
    }
    override fun getColumnByTitle(title: String): DataTableColumn {
        return CalendarTableColumn.values().filter { it.title == title }.first()
    }
}